backend/src/index.js

```diff
@@ -91,9 +97,151 @@ async function main() {
+    wamp.session.publish(uri, args);
+  };
+
+  const wampSqlSelectQueryCount = async (args) => {
+    const uri = `com.nearprotocol.${wampNearNetworkName}.explorer.select`;
+    const res = await wamp.session.call(uri, args);
+    return res[0];
+  };
+
+  const wampSqlSelectQueryRows = async (args) => {
+    const uri = `com.nearprotocol.${wampNearNetworkName}.explorer.select`;
+    return await wamp.session.call(uri, args);
+  };
+
+  // regular check finalTimesamp and publish to final-timestamp uri
+  const regularCheckFinalTimestamp = async () => {
+    try {
+      if (wamp.session) {
+        const finalTimestamp = await queryFinalTimestamp();
+        wampPublish("final-timestamp", [finalTimestamp]);
+      }
+    } catch (error) {
+      console.warn("Regular querying RPC crashed due to:", error);
+    }
+    setTimeout(regularCheckFinalTimestamp, regularQueryRPCInterval);
+  };
+  setTimeout(regularCheckFinalTimestamp, 0);
+
+  // regular check block/tx data stats and publish to data-stats uri
+  const totalStats = async () => {
+    const [
+      totalBlocks,
+      totalTransactions,
+      totalAccounts,
+      lastDayTxCount,
+      lastBlockHeight,
+    ] = await Promise.all([
+      wampSqlSelectQueryCount([`SELECT COUNT(*) as total FROM blocks`]),
+      wampSqlSelectQueryCount([`SELECT COUNT(*) as total FROM transactions`]),
+      wampSqlSelectQueryCount([`SELECT COUNT(*) as total FROM accounts`]),
+      wampSqlSelectQueryCount([
+        `SELECT COUNT(*) as total FROM transactions
+        WHERE block_timestamp > (strftime('%s','now') - 60 * 60 * 24) * 1000`,
+      ]),
+      wampSqlSelectQueryCount([
+        `SELECT height FROM blocks ORDER BY height DESC LIMIT 1`,
+      ]),
+    ]);
+    return {
+      totalAccounts: totalAccounts.total,
+      totalBlocks: totalBlocks.total,
+      totalTransactions: totalTransactions.total,
+      lastDayTxCount: lastDayTxCount.total,
+      lastBlockHeight: lastBlockHeight.height,
+    };
+  };
+
+  const regularCheckDataStats = async () => {
+    try {
+      if (wamp.session) {
+        const dataStats = await totalStats();
+        wampPublish("data-stats", [{ dataStats }]);
+      }
+    } catch (error) {
+      console.warn("Regular querying data stats crashed due to:", error);
+    }
+    setTimeout(regularCheckDataStats, regularQueryStatsInterval);
+  };
+  setTimeout(regularCheckDataStats, 0);
+
+  // regular check node status and publish to nodes uri
+  const addNodeInfo = async (nodes) => {
+    const accountArray = nodes
+      .map((node) => '"' + node.account_id + '"')
+      .join(",");
+    let nodesInfo = await wampSqlSelectQueryRows([
+      `SELECT ip_address as ipAddress, account_id as accountId, node_id as nodeId, 
+        last_seen as lastSeen, last_height as lastHeight,status,
+        agent_name as agentName, agent_version as agentVersion, agent_build as agentBuild
+              FROM nodes
+              WHERE account_id IN (${accountArray})
```

Comment:
You may easily introduce security issues when you construct SQL queries with inline values. Use named replacement here.

Reply:
I tried: 
![image](https://user-images.githubusercontent.com/33007476/88978249-324a7080-d274-11ea-89c3-518c69d676bb.png)

but sql query does not deal with it  correctly and gives this:
![image](https://user-images.githubusercontent.com/33007476/88978329-58701080-d274-11ea-9799-b620750a498f.png)
it have a ' ' out of string and incorrect

Reply:
You should just pass a JS array of account IDs instead of a comma-separated string.